framework/src/controller/migrations/migration.js

```diff
@@ -253,31 +274,82 @@ class Migration extends BaseEntity {
 						migration && {
 							id: migration[1],
 							name: migration[2],
-							path: path.join('../sql/migrations/updates', migrationFile),
+							path: `${updatesPath}/${migrationFile}`,
 						}
 					);
 				})
 				.sort((a, b) => a.id - b.id) // Sort by migration ID, ascending
 				.filter(
 					migration =>
 						migration &&
-						fs
-							.statSync(path.join(this.sqlDirectory, migration.path))
-							.isFile() &&
+						fs.statSync(migration.path).isFile() &&
 						(!lastMigrationId || +migration.id > lastMigrationId)
 				)
-				.map(f => {
-					f.file = this.adapter.loadSQLFile(f.path, this.sqlDirectory);
-					return f;
+				.map(migration => {
+					migration.file = this.adapter.loadSQLFile(migration.path, '');
+					return migration;
+				})
+		);
+	}
+
+	/**
+	 * Reads/updates' folder and returns an array of objects for further processing.
+	 *
+	 * @param {number} lastMigrationId
+	 * @returns {Promise<Array<Object>>}
+	 * Promise object that resolves with an array of objects `{id, name, path, file}`.
+	 */
+	async readPending(migrationsObj, savedMigrations) {
+		// const updatesPath = `${this.sqlDirectory}/updates`;
+		return Object.keys(migrationsObj).reduce((prev, namespace) => {
+			const curr = migrationsObj[namespace]
+				.map(migrationFile => {
+					const migration = migrationFile.match(/(\d+)_(.+).sql/);
+					return (
+						migration && {
```

Comment:
not sure about this syntax, what does it do?
is it suppose to be like this?
```
{
   ...migration,
   id:
}
```

Reply:
It returns the object if `migration` is true.
We can't use destructuring here because `migration` will be an array and we want to build an object.

Reply:
I think `map` should change the input to output one to one.
so in `true` case it's fine, but in `false` case it will be undefined. (probably needs flatMap in this case)

Reply:
It's still fine because right after we're filtering the map result